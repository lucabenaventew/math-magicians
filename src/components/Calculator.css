.wrapper {
  width: fit-content;
}

#input {
  width: calc(100% - 10px);
  line-height: 30px;
  border: none;
  background-color: rgb(104, 104, 104);
  padding: 5px;
  color: #fff;
  text-align: right;
  font-size: 2vh;
}

.keyboard-wrapper {
  width: 100%;
  background-color: rgb(184, 184, 184);
}

.keyboard button {
  aspect-ratio: 3/2;
  height: 8vh;
  font-size: 2vh;
  width: 100%;
  border: none;
  cursor: pointer;
}

.keyboard th:nth-child(4) button {
  background-color: orange;
}

.equal {
  background-color: orange;
}
