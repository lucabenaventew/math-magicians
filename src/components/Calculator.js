import React from 'react';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  render() {
    return (
      <div className="wrapper">
        <form>
          <input type="text" id="input" />
        </form>
        <table className="keyboard-wrapper">
          <tbody className="keyboard">
            <tr>
              <th><button type="button" className="AC">AC</button></th>
              <th><button type="button" className="plus-minus">+/-</button></th>
              <th><button type="button" className="percent">%</button></th>
              <th><button type="button" className="divide">:</button></th>
            </tr>
            <tr>
              <th><button type="button" className="one">1</button></th>
              <th><button type="button" className="two">2</button></th>
              <th><button type="button" className="three">3</button></th>
              <th><button type="button" className="multiply">X</button></th>
            </tr>
            <tr>
              <th><button type="button" className="four">4</button></th>
              <th><button type="button" className="five">5</button></th>
              <th><button type="button" className="six">6</button></th>
              <th><button type="button" className="minus">-</button></th>
            </tr>
            <tr>
              <th><button type="button" className="seven">7</button></th>
              <th><button type="button" className="eight">8</button></th>
              <th><button type="button" className="nine">9</button></th>
              <th><button type="button" className="plus">+</button></th>
            </tr>
            <tr>
              <th colSpan={2}><button type="button" className="zero">0</button></th>
              <th><button type="button" className="dot">.</button></th>
              <th><button type="button" className="equal">=</button></th>
            </tr>
          </tbody>
        </table>
      </div>
    );
  }
}

export default Calculator;
