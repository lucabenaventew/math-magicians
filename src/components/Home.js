import React from 'react';

function Home() {
  return (
    <div className="home">
      <h3>Welcome to Math Magicians!</h3>
      <p>
        Math Magicians is a website were you will be able to use a calculator and read a quote
        said by Albert Einstein.
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione
        minima, maxime repudiandae quo possimu svoluptatum natus, voluptates
        quae, qui pariatur itaque perspiciat isdolores iusto? Soluta asperiores
        aperiam possimus iste id perspicia tisdolores iusto? Soluta asperiores.
      </p>
    </div>
  );
}
export default Home;
