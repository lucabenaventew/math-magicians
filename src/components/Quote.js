import React from 'react';

function Quote() {
  return (
    <div>
      <p className="quote">
        Pure mathematics is, in its way, the poetry of logical ideas.
        ~ Albert Einstein
      </p>
    </div>
  );
}
export default Quote;
