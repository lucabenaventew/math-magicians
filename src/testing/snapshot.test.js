import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Quote from '../components/Quote';

describe('Working with snapshots', () => {
  test('render Home', () => {
    const home = renderer.create(<Home />).toJSON();
    expect(home).toMatchSnapshot();
  });
  test('render Home', () => {
    const calculator = renderer.create(<Calculator />).toJSON();
    expect(calculator).toMatchSnapshot();
  });
  test('render Home', () => {
    const quote = renderer.create(<Quote />).toJSON();
    expect(quote).toMatchSnapshot();
  });
});
