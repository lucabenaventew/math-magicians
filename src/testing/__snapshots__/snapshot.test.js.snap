// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Working with snapshots render Home 1`] = `
<div
  className="home"
>
  <h3>
    Welcome to Math Magicians!
  </h3>
  <p>
    Math Magicians is a website were you will be able to use a calculator and read a quote said by Albert Einstein.
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione minima, maxime repudiandae quo possimu svoluptatum natus, voluptates quae, qui pariatur itaque perspiciat isdolores iusto? Soluta asperiores aperiam possimus iste id perspicia tisdolores iusto? Soluta asperiores.
  </p>
</div>
`;

exports[`Working with snapshots render Home 2`] = `
<div
  className="calculator-wrapper"
>
  <h2
    className="calculator-title"
  >
    Lets do some Math!
  </h2>
  <div
    className="calculator"
  >
    <form>
      <input
        id="input"
        name="input"
        type="text"
      />
    </form>
    <table
      className="keyboard-wrapper"
    >
      <tbody
        className="keyboard"
      >
        <tr>
          <th>
            <button
              className="AC"
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
          </th>
          <th>
            <button
              className="plus-minus"
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
          </th>
          <th>
            <button
              className="percent"
              onClick={[Function]}
              type="button"
            >
              %
            </button>
          </th>
          <th>
            <button
              className="divide"
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
          </th>
        </tr>
        <tr>
          <th>
            <button
              className="one"
              onClick={[Function]}
              type="button"
            >
              1
            </button>
          </th>
          <th>
            <button
              className="two"
              onClick={[Function]}
              type="button"
            >
              2
            </button>
          </th>
          <th>
            <button
              className="three"
              onClick={[Function]}
              type="button"
            >
              3
            </button>
          </th>
          <th>
            <button
              className="multiply"
              onClick={[Function]}
              type="button"
            >
              x
            </button>
          </th>
        </tr>
        <tr>
          <th>
            <button
              className="four"
              onClick={[Function]}
              type="button"
            >
              4
            </button>
          </th>
          <th>
            <button
              className="five"
              onClick={[Function]}
              type="button"
            >
              5
            </button>
          </th>
          <th>
            <button
              className="six"
              onClick={[Function]}
              type="button"
            >
              6
            </button>
          </th>
          <th>
            <button
              className="minus"
              onClick={[Function]}
              type="button"
            >
              -
            </button>
          </th>
        </tr>
        <tr>
          <th>
            <button
              className="seven"
              onClick={[Function]}
              type="button"
            >
              7
            </button>
          </th>
          <th>
            <button
              className="eight"
              onClick={[Function]}
              type="button"
            >
              8
            </button>
          </th>
          <th>
            <button
              className="nine"
              onClick={[Function]}
              type="button"
            >
              9
            </button>
          </th>
          <th>
            <button
              className="plus"
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </th>
        </tr>
        <tr>
          <th
            colSpan={2}
          >
            <button
              className="zero"
              onClick={[Function]}
              type="button"
            >
              0
            </button>
          </th>
          <th>
            <button
              className="dot"
              onClick={[Function]}
              type="button"
            >
              .
            </button>
          </th>
          <th>
            <button
              className="equal"
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Working with snapshots render Home 3`] = `
<div>
  <p
    className="quote"
  >
    Pure mathematics is, in its way, the poetry of logical ideas. ~ Albert Einstein
  </p>
</div>
`;
